presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewLogicalPlanner.java

```diff
@@ -1077,6 +1089,551 @@ public void testBaseToViewConversionWithGroupBy()
+                                                                                    ImmutableList.of(),
+                                                                                    ImmutableMap.of(),
+                                                                                    Optional.empty(),
+                                                                                    PARTIAL,
+                                                                                    node(
+                                                                                            ProjectNode.class, constrainedTableScan(
+                                                                                                    table,
+                                                                                                    ImmutableMap.of("ds", singleValue(createVarcharType(10), utf8Slice("2021-07-12"))),
+                                                                                                    ImmutableMap.of("ds", "ds"))))))))),
+                                    anyTree(
+                                            constrainedTableScan(
+                                                    view,
+                                                    ImmutableMap.of("ds", singleValue(createVarcharType(10), utf8Slice("2021-07-11"))),
+                                                    ImmutableMap.of())))));
+
+            assertPlan(getSession(), viewQuery, expectedPattern);
+            assertPlan(queryOptimizationWithMaterializedView, baseQuery, expectedPattern);
+        }
+        finally {
+            queryRunner.execute("DROP MATERIALIZED VIEW IF EXISTS " + view);
+            queryRunner.execute("DROP TABLE IF EXISTS " + table);
+        }
+    }
+
+    @Test
+    public void testBaseToViewConversionCountOutputAsExpected()
+    {
+        Session queryOptimizationWithMaterializedView = Session.builder(getSession())
+                .setSystemProperty(QUERY_OPTIMIZATION_WITH_MATERIALIZED_VIEW_ENABLED, "true")
+                .build();
+        QueryRunner queryRunner = getQueryRunner();
+
+        String table = "my_table";
+        String view = "my_view";
+
+        try {
+            queryRunner.execute(format("CREATE TABLE %s" +
+                            "(a_field bigint, grouping_key bigint, ds varchar(10)) " +
+                            "WITH (partitioned_by=array['ds'])",
+                    table));
+
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s " +
+                            "WITH (partitioned_by = ARRAY['ds']) " +
+                            "AS SELECT COUNT(*) AS a_count, grouping_key, ds " +
+                            "FROM %s " +
+                            "GROUP BY ds, grouping_key\n",
+                    view, table));
+
+            assertTrue(getQueryRunner().tableExists(getSession(), view));
+
+            assertUpdate(format("INSERT INTO %s (a_field, grouping_key, ds) " +
```

Comment:
You can do this single insert statement.

INSERT INTO t (a,b,c) VALUES ('a', 'b', 'c'), ('a2', 'b2', 'c2'), ...;